packages/frontend-app-api/src/wiring/parameters.ts

```diff
@@ -196,18 +196,43 @@ export function mergeExtensionParameters(options: {
 }): ExtensionInstanceParameters[] {
   const { sources, builtinExtensions, parameters } = options;
 
+  const pluginExtensionIds = new Set<string>();
+  const pluginExtensions = sources.flatMap(source => {
+    return source.extensions
+      .filter(extension => {
+        // Filter out duplicated extensions
+        const id = extension.id;
+        if (!pluginExtensionIds.has(id)) {
+          pluginExtensionIds.add(id);
+          return true;
+        }
+        return false;
```

Comment:
What do you think about making this a throw error instead? I'm feeling we want to be very strict here, at least to begin with. That way we spot issues here earlier and can iron them out. Handling this gracefully I feel is more likely to cause trouble.

I'd even go as far as thinking that we should apply this check to the `overrides` array below, so that includes the built-in extensions. It might sound a bit strange now because there's no other way to do it, but I'm pretty sure we don't want plugins to be able to override core extensions. We'd instead have some other form of feature that provides extension overrides that we haven't built just yet.

Reply:
Make sense, code updated 🙂 : https://github.com/backstage/backstage/pull/20032/commits/c71da2a7907c53e6d147b76925ee0f1341406669